lms/djangoapps/shoppingcart/sync.py

```diff
@@ -0,0 +1,106 @@
+"""
+This file exposes methods to initiate a transaction synchronization from a remote
+Payment Processor and the local database
+"""
+
+import pytz
+from django.conf import settings
+from django.core.mail.message import EmailMessage
+from datetime import datetime
+from django.utils.translation import ugettext as _
+
+from .models import (
+    Order,
+    PaymentTransactionSync,
+    PaymentTransaction,
+)
+from .processors import (
+    synchronize_transactions
+)
+
+
+def perform_sync(start_date=None, end_date=None, summary_email_to=None):
+    """
+    This method will perform a sync from the Payment Processor to the local database in the given date ranges. If no
+    dates are provided, then we will sync from the last known sync date that has been stored in our sync history
+    records. If there has been no sync history yet, we will query the local database to see the first purchased
+    Order and use that as the sync date.
+
+    Note that we only do synchronizations on full days
+    """
+
+    if not start_date:
+        # look in the list of synched transactions that we have already done
+        start_date = PaymentTransaction.get_last_processed_date()
+
+        if not start_date:
+            order = (Order.objects.all().order_by('id'))[0]
+            start_date = order.purchase_time
+            print start_date
```

Comment:
Actually, after some HipChatting with @jab5569 I might put this back on hold and plan to use the warehouse that will be developed by his team.